src/services/completions.ts

```diff
@@ -1153,7 +1249,7 @@ namespace ts.Completions {
             Debug.assertEachIsDefined(tagSymbols, "getJsxIntrinsicTagNames() should all be defined");
             tryGetGlobalSymbols();
             symbols = tagSymbols.concat(symbols);
-            completionKind = CompletionKind.MemberLike;
+            completionKind = CompletionKind.Global;

```

Comment:
AFAICT this `completionKind` had no effect prior to me moving the aforementioned filtering code, but was conceptually wrong, and changing it to be conceptually right allowed me to move the filter.